RecoCTPPS/PixelLocal/src/RPixPlaneCombinatoryTracking.cc

```diff
@@ -62,21 +62,21 @@ void RPixPlaneCombinatoryTracking::initialize(){
 //------------------------------------------------------------------------------------------------//
     
 //This function produces all the possible plane combinations extracting numberToExtract planes over numberOfPlanes planes
-RPixPlaneCombinatoryTracking::PlaneCombinations
-RPixPlaneCombinatoryTracking::getPlaneCombinations(const std::vector<uint32_t> &inputPlaneList, uint32_t numberToExtract)
+const RPixPlaneCombinatoryTracking::PlaneCombinations &
+RPixPlaneCombinatoryTracking::getPlaneCombinations(const std::vector<uint32_t> &inputPlaneList, uint32_t numberToExtract) const
 {
     uint32_t numberOfPlanes = inputPlaneList.size();
     std::string bitmask(numberToExtract, 1); // numberToExtract leading 1's
     bitmask.resize(numberOfPlanes, 0); // numberOfPlanes-numberToExtract trailing 0's
-    PlaneCombinations planeCombinations;
+    static PlaneCombinations planeCombinations;
+    planeCombinations.clear();
```

Comment:
this is not thread-safe.

Either planeCombinations is to be moved to data members to be able to return it (but then the method can not be const-qualified), or the interface should be changed to take ```PlaneCombinations& ``` by argument; or we pay the price of copying the results out on return.